tests/filecheck/transforms/lower-csl-wrapper.mlir

```diff
@@ -153,29 +156,124 @@ builtin.module {
+// CONST-NEXT:     "csl.export"() <{"var_name" = @gauss_seidel_func, "type" = () -> ()}> : () -> ()
+// CONST-NEXT:     csl.func @gauss_seidel_func() {
+// CONST-NEXT:       %scratchBuffer = memref.alloc() {"alignment" = 64 : i64} : memref<510xf32>
+// CONST-NEXT:       csl_stencil.apply(%inputArr : memref<512xf32>, %scratchBuffer : memref<510xf32>) outs (%outputArr : memref<512xf32>) <{"bounds" = #stencil.bounds<[0, 0], [1, 1]>, "num_chunks" = 2 : i64, "operandSegmentSizes" = array<i32: 1, 1, 0, 0, 1>, "swaps" = [#csl_stencil.exchange<to [1, 0]>, #csl_stencil.exchange<to [-1, 0]>, #csl_stencil.exchange<to [0, 1]>, #csl_stencil.exchange<to [0, -1]>], "topo" = #dmp.topo<1022x510>}> ({
+// CONST-NEXT:       ^0(%arg2 : memref<4x255xf32>, %arg3 : index, %arg4 : memref<510xf32>):
+// CONST-NEXT:         %0 = csl_stencil.access %arg2[1, 0] : memref<4x255xf32>
+// CONST-NEXT:         %1 = csl_stencil.access %arg2[-1, 0] : memref<4x255xf32>
+// CONST-NEXT:         %2 = csl_stencil.access %arg2[0, 1] : memref<4x255xf32>
+// CONST-NEXT:         %3 = csl_stencil.access %arg2[0, -1] : memref<4x255xf32>
+// CONST-NEXT:         %4 = memref.subview %arg4[%arg3] [255] [1] : memref<510xf32> to memref<255xf32, strided<[1], offset: ?>>
+// CONST-NEXT:         "csl.fadds"(%4, %3, %2) : (memref<255xf32, strided<[1], offset: ?>>, memref<255xf32>, memref<255xf32>) -> ()
+// CONST-NEXT:         "csl.fadds"(%4, %4, %1) : (memref<255xf32, strided<[1], offset: ?>>, memref<255xf32, strided<[1], offset: ?>>, memref<255xf32>) -> ()
+// CONST-NEXT:         "csl.fadds"(%4, %4, %0) : (memref<255xf32, strided<[1], offset: ?>>, memref<255xf32, strided<[1], offset: ?>>, memref<255xf32>) -> ()
+// CONST-NEXT:         csl_stencil.yield %arg4 : memref<510xf32>
+// CONST-NEXT:       }, {
+// CONST-NEXT:       ^1(%arg2_1 : memref<512xf32>, %arg3_1 : memref<510xf32>):
+// CONST-NEXT:         %5 = memref.subview %arg2_1[2] [510] [1] : memref<512xf32> to memref<510xf32, strided<[1], offset: 2>>
+// CONST-NEXT:         %6 = memref.subview %arg2_1[0] [510] [1] : memref<512xf32> to memref<510xf32, strided<[1]>>
+// CONST-NEXT:         "csl.fadds"(%arg3_1, %arg3_1, %6) : (memref<510xf32>, memref<510xf32>, memref<510xf32, strided<[1]>>) -> ()
+// CONST-NEXT:         "csl.fadds"(%arg3_1, %arg3_1, %5) : (memref<510xf32>, memref<510xf32>, memref<510xf32, strided<[1], offset: 2>>) -> ()
+// CONST-NEXT:         %7 = arith.constant 1.666600e-01 : f32
+// CONST-NEXT:         "csl.fmuls"(%arg3_1, %arg3_1, %7) : (memref<510xf32>, memref<510xf32>, f32) -> ()
+// CONST-NEXT:         csl_stencil.yield %arg3_1 : memref<510xf32>
+// CONST-NEXT:       }) to <[0, 0], [1, 1]>
+// CONST-NEXT:       csl.return
+// CONST-NEXT:     }
+// CONST-NEXT:     %0 = "csl.member_access"(%memcpyMod_2) <{"field" = "LAUNCH"}> : (!csl.imported_module) -> !csl.color
+// CONST-NEXT:     "csl.rpc"(%0) : (!csl.color) -> ()
+// CONST-NEXT:   }) {"sym_name" = "gauss_seidel_func_program"} : () -> ()
+// CONST-NEXT: }
+// CONST-EMPTY:
```

Comment:
This test case is huge. Do we need such a large test case to test this feature? Or can there be a smaller minimal test case?

Reply:
I added a shorter test, with and without the flag. @dk949